File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn dna-nucleotide-complement
  [dna-nucleotide]
  {:pre [(.contains [\C \G \A \T] dna-nucleotide)]}
  (case dna-nucleotide
    \C \G
    \G \C
    \A \U
    \T \A))

(defn to-rna [dna]
  (apply str (map dna-nucleotide-complement dna)))

File: go/rna-transcription/rna_transcription.go
// Package strand implements the solution to the strand exercise on exercism
package strand

type nucleotide = rune

var DNAComplement = map[nucleotide]nucleotide{
	'A': 'U',
	'C': 'G',
	'T': 'A',
	'G': 'C',
}

// ToRNA returns the RNA complement of the given dna
func ToRNA(dna string) string {
	result := make([]nucleotide, len(dna))

	for i, n := range dna {
		result[i] = DNAComplement[n]
	}

	return string(result)
}
